File: Data Structures & Algorithms/jump-game/submission-4.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int currGoal = nums.size() - 1;

        for (int i = nums.size() - 2; i > -1; i--)
        {
            if (i + nums[i] >= currGoal)
            {
                currGoal = i;
            }
        }

        return currGoal == 0;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-4.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        integerLocDict = {}

        for i in range(len(nums)):
            curr = nums[i]

            diff = target - curr
            if diff in integerLocDict:
                return [integerLocDict[diff], i]

            integerLocDict[curr] = i

        return []